aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIOProperties.java

```diff
@@ -819,6 +820,14 @@ public <T extends S3ClientBuilder> void applyS3AccessGrantsConfigurations(T buil
     }
   }
 
+  private static final String S3_FILE_IO_SIGNATURE = "s3fileio/" + EnvironmentContext.get();
```

Comment:
this should be defined at the top of the file (after all the other `public static final` vars) . Also signature seems rather confusing to me. Why not just use `S3_FILE_IO_USER_AGENT`?

Reply:
Ack, thanks! In my mind signature/fingerprint a software leaves after it touched something mean the same thing, but the term signature also has uses in cryptography and AuthN so I get the confusion. 

I agree that user agent will be more descriptive here, changed it now.